埋め込みファイルの数が多いとARMHFでビルドエラーが出る

## 不具合の内容

埋め込みファイルの数が多いとarmhfビルドがエラーになります。

### 現象・ログ

実際のビルドログはこちらです https://github.com/Hiroshiba/voicevox_core/runs/6438083012?check_suite_focus=true

一部抜粋してみます。

```bash
Run cmake --build . --config Release

...

/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: BFD (GNU Binutils for Ubuntu) 2.30 assertion fail ../../bfd/elf32-arm.c:9509
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: BFD (GNU Binutils for Ubuntu) 2.30 assertion fail ../../bfd/elf32-arm.c:9509
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: BFD (GNU Binutils for Ubuntu) 2.30 assertion fail ../../bfd/elf32-arm.c:9509
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: BFD (GNU Binutils for Ubuntu) 2.30 assertion fail ../../bfd/elf32-arm.c:9509
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: BFD (GNU Binutils for Ubuntu) 2.30 assertion fail ../../bfd/elf32-arm.c:9509
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: BFD (GNU Binutils for Ubuntu) 2.30 assertion fail ../../bfd/elf32-arm.c:9509
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: BFD (GNU Binutils for Ubuntu) 2.30 assertion fail ../../bfd/elf32-arm.c:9509
collect2: error: ld returned 1 exit status
make[2]: *** [core/libcore.so] Error 1
make[1]: *** [core/CMakeFiles/core.dir/all] Error 2
make: *** [all] Error 2
core/CMakeFiles/core.dir/build.make:829: recipe for target 'core/libcore.so' failed
CMakeFiles/Makefile2:115: recipe for target 'core/CMakeFiles/core.dir/all' failed
Makefile:135: recipe for target 'all' failed
Error: Process completed with exit code 2.
```

### 再現手順

ブランチは[こちら](https://github.com/Hiroshiba/voicevox_core/tree/%E5%9F%8B%E3%82%81%E8%BE%BC%E3%81%BF%E3%83%95%E3%82%A1%E3%82%A4%E3%83%AB%E3%81%AE%E6%95%B0%E3%81%8C%E5%A4%9A%E3%81%84%E3%81%A8ARMHF%E3%81%A7%E3%83%93%E3%83%AB%E3%83%89%E3%82%A8%E3%83%A9%E3%83%BC%E3%81%8C%E5%87%BA%E3%82%8B%E3%81%8B%E3%82%82)です

* DECODEなどのモデルを大量にembedする
  * こんな感じです https://github.com/Hiroshiba/voicevox_core/commit/d6a7dd831ed96f46a47501d94013ea3909f8946b
* ビルドする
* エラーが出る

### 期待動作

エラーが出ずにビルドできる

## OSの種類/ディストリ/バージョン

linux armhf

## その他

* モデル埋め込みのプルリクエスト
  * https://github.com/VOICEVOX/voicevox_core/pull/99
* armhf周りのプルリクエスト
  * https://github.com/VOICEVOX/voicevox_core/pull/41

新しいキャラクター追加のためにモデルを足したところエラーに遭遇しました。
解決が難しそうであれば、いったんarmhfをビルドから外したいと思います。

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -96,6 +96,7 @@ jobs:
             cc_version: "8"
             cxx_version: "8"
             arch: arm-linux-gnueabihf
+            cmake_additional_options: -DARMHF=ON
 
           - os: ubuntu-18.04
             device: cpu-arm64

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,6 +26,12 @@ if(DIRECTML)
   determine_target_architecture(DML_ARCH)
 endif()
 
+option(ARMHF "Enables building for ARMHF" OFF)
+if(ARMHF)
+  # https://github.com/VOICEVOX/voicevox_core/issues/132
+  add_link_options("-Xlinker" "--long-plt")
+endif()
+
 add_subdirectory(core)
 add_subdirectory(open_jtalk/src)
 

